Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs

```diff
@@ -1067,20 +1073,38 @@ private Declaration FindModuleScopeDeclaration(string identifierName, Declaratio
                 localScope = _currentScope;
             }
 
+            if (localScope.DeclarationType == DeclarationType.Project)
+            {
+                return null;
+            }
+
             if (identifierName == "Me" && _moduleDeclaration.DeclarationType == DeclarationType.Class)
             {
                 return _moduleDeclaration;
             }
 
-            var matches = _declarationFinder.MatchName(identifierName);
+            var scope = localScope; // avoid implicitly capturing 'this'
+            var matches = _declarationFinder.MatchName(identifierName).Where(item => !item.Equals(scope)).ToList();
+
             var result = matches.Where(item =>
-                item.ParentScope == localScope.ParentScope
+                (localScope.ParentScopeDeclaration == null || localScope.ParentScopeDeclaration.Equals(item.ParentScopeDeclaration))
                 && !item.DeclarationType.HasFlag(DeclarationType.Member)
                 && !_moduleTypes.Contains(item.DeclarationType)
-                && item.DeclarationType != DeclarationType.UserDefinedType
+                && item.DeclarationType != DeclarationType.UserDefinedType && item.DeclarationType != DeclarationType.Enumeration
                 && (item.DeclarationType != DeclarationType.Event || IsLocalEvent(item, localScope)))
```

Comment:
THis huge predicate could use a name... at least in the diff I can't understand it... maybe the surrounding method gives more information :/